pkg/template/v2/template.go

```diff
@@ -90,39 +92,91 @@ func init() {
 	})
 }
 
-func applyToTarget(k string, val []byte, target esapi.TemplateTarget, secret *corev1.Secret) {
+func applyToTarget(k string, val []byte, target string, obj client.Object) error {
+	target = strings.ToLower(target)
 	switch target {
-	case esapi.TemplateTargetAnnotations:
-		if secret.Annotations == nil {
-			secret.Annotations = make(map[string]string)
+	case "annotations":
+		annotations := obj.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
 		}
-		secret.Annotations[k] = string(val)
-	case esapi.TemplateTargetLabels:
-		if secret.Labels == nil {
-			secret.Labels = make(map[string]string)
+		annotations[k] = string(val)
+		obj.SetAnnotations(annotations)
+	case "labels":
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
 		}
-		secret.Labels[k] = string(val)
-	case esapi.TemplateTargetData:
-		if secret.Data == nil {
-			secret.Data = make(map[string][]byte)
+		labels[k] = string(val)
+		obj.SetLabels(labels)
+	case "data":
```

Comment:
what if someone sets string data on a configmap? whats the behavior?

Reply:
It will probably fail horribly. :D :D I have to check that as well. :) I assume ( hopefully ) the api would throw back an error that `spec` ( or any other field ) is not allowed on the given object.